### PR TITLE
[modules] airspeed UAVCAN

### PR DESCRIPTION
**Pull request by dewagter**
ABI

### CODE REVIEW

File: sw/airborne/modules/sensors/airspeed_uavcan.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  //float static_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  //canardDecodeScalar(transfer, (uint32_t)88, 16, false, (void*)&tmp_float);
+  //float diff_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  canardDecodeScalar(transfer, (uint32_t)104, 16, false, (void*)&tmp_float);
+  float static_air_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  //canardDecodeScalar(transfer, (uint32_t)120, 16, false, (void*)&tmp_float);
+  //float pitot_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+
+  if(!isnan(diff_p)) {
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
```
Comment (fvantienen) [inline sw/airborne/modules/sensors/airspeed_uavcan.c:75]: Change to `AIRSPEED_UAVCAN_LOWPASS_FILTER`

File: sw/airborne/modules/sensors/airspeed_uavcan.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  if(!isnan(static_air_temp))
+    //AbiSendMsgTEMPERATURE(UAVCAN_SENDER_ID, static_air_temp);
+    airspeed_uavcan.temperature = static_air_temp;
+}
+
+void airspeed_uavcan_init(void)
+{
+  // Setup low pass filter with time constant and 100Hz sampling freq
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
+  init_butterworth_2_low_pass(&airspeed_filter, MS45XX_LOWPASS_TAU, MS45XX_I2C_PERIODIC_PERIOD, 0);
```
Comment (fvantienen) [inline sw/airborne/modules/sensors/airspeed_uavcan.c:94]: Change the `TAU` and `PERIOD`, I guess the `PERIOD` should be a simple define and add both to the doc of the module.

File: sw/airborne/modules/sensors/airspeed_uavcan.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  //canardDecodeScalar(transfer, (uint32_t)120, 16, false, (void*)&tmp_float);
+  //float pitot_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+
+  if(!isnan(diff_p)) {
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
+    float diff_p_filt = update_butterworth_2_low_pass(&airspeed_filter, diff_p);
+    airspeed_uavcan.diff_p = diff_p;
+    //AbiSendMsgBARO_DIFF(UAVCAN_SENDER_ID, diff_p_filt);
+#else
+    //AbiSendMsgBARO_DIFF(UAVCAN_SENDER_ID, diff_p);
```
Comment (fvantienen) [inline sw/airborne/modules/sensors/airspeed_uavcan.c:80]: Enable the ABI messages by default and maybe add a define to disable them.

**Review by fvantienen — CHANGES_REQUESTED** [review]

File: sw/airborne/modules/sensors/airspeed_uavcan.c (hunk 1 above)
Comment (dewagter) [reply sw/airborne/modules/sensors/airspeed_uavcan.c]: done

File: sw/airborne/modules/sensors/airspeed_uavcan.c (hunk 2 above)
Comment (dewagter) [reply sw/airborne/modules/sensors/airspeed_uavcan.c]: done

**Review by fvantienen — APPROVED** [review]

**Pull request merged**